### PR TITLE
Add Snippets for working with Assets in Cloud Security Command Center.

### PR DESCRIPTION
**Pull request by emkornfield**
- I missed instruction for how to add a new key, but I believe
a new account is needed, so prestaged assets can be queried.

Fixes #<issue_number_goes_here> (it's a good idea to open an issue first for context and/or discussion)

### CODE REVIEW

File: .kokoro/presubmit/securitycenter-it.cfg [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Format: //devtools/kokoro/config/proto/build.proto
```
Comment (kolea2) [inline .kokoro/presubmit/securitycenter-it.cfg]: This file is to configure running integration tests found in the google-cloud-clients directory. Is that the intent here, or is this just for samples?

File: google-cloud-examples/src/main/java/com/google/cloud/examples/securitycenter/snippets/AssetSnippets.java [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+ */
+public class AssetSnippets {
+
+  private final SecurityCenterClient securityCenterClient;
+  private final OrganizationName organizationName;
+
+  /**
+   * Filter that returns all projects in the organization
+   */
+  // [START  asset_resource_project_filter]
```
Comment (kolea2) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/securitycenter/snippets/AssetSnippets.java]: extra whitespace after START and END

File: google-cloud-examples/src/main/java/com/google/cloud/examples/securitycenter/snippets/AssetSnippets.java [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+   *
+   * @param filter The filter that assets must meet (e.g. {@link #PROJECT_ASSET_FILTERS}).  If null,
+   * all assets in the organization are returned.
+   * @param asOf The instant in time to query for.  If null, current time is assumed
+   */
+
+  // [TARGET list_assets
+  // [VARIABLE "filter"]
+  // [VARIABLE "as_of"]
+  // [START list_assets ]
```
Comment (kolea2) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/securitycenter/snippets/AssetSnippets.java]: whitespace

File: google-cloud-examples/src/main/java/com/google/cloud/examples/securitycenter/snippets/AssetSnippets.java [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+  public ImmutableList<ListAssetsResult> listAssets(String filter,
+      Instant asOf) {
+    ListAssetsRequest.Builder request =
+        ListAssetsRequest.newBuilder().setParent(organizationName.toString());
+    //
```
Comment (kolea2) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/securitycenter/snippets/AssetSnippets.java]: missing comment?

File: google-cloud-examples/src/main/java/com/google/cloud/examples/securitycenter/snippets/AssetSnippets.java [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+    // the Iterable returned response.iterateAll() directly.
+    return ImmutableList.copyOf(response.iterateAll());
+  }
+  // [END list_assets ]
+
+  /**
+   * Run and print results from common queries.
+   */
+  void demoListAssets() {
+    // [ START demo_list_assets ] ]
```
Comment (kolea2) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/securitycenter/snippets/AssetSnippets.java]: clean up whitespace and ]

Comment (kolea2) [reply google-cloud-examples/src/main/java/com/google/cloud/examples/securitycenter/snippets/AssetSnippets.java]: (there are a few of the same cleanups required in this file)

File: google-cloud-examples/src/main/java/com/google/cloud/examples/securitycenter/snippets/AssetSnippets.java (hunk 2 above)
Comment (emkornfield) [reply google-cloud-examples/src/main/java/com/google/cloud/examples/securitycenter/snippets/AssetSnippets.java]: Fixed

File: google-cloud-examples/src/main/java/com/google/cloud/examples/securitycenter/snippets/AssetSnippets.java [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+  /**
+   * Lists assets for an organization given meeting <code>filter</code> as of a specific instant in
+   * time.
+   *
+   * @param filter The filter that assets must meet (e.g. {@link #PROJECT_ASSET_FILTERS}).  If null,
+   * all assets in the organization are returned.
+   * @param asOf The instant in time to query for.  If null, current time is assumed
+   */
+
+  // [TARGET list_assets
```
[the remark this answers is not in the document]
Comment (emkornfield) [reply google-cloud-examples/src/main/java/com/google/cloud/examples/securitycenter/snippets/AssetSnippets.java]: Fixed.

File: google-cloud-examples/src/main/java/com/google/cloud/examples/securitycenter/snippets/AssetSnippets.java (hunk 3 above)
Comment (emkornfield) [reply google-cloud-examples/src/main/java/com/google/cloud/examples/securitycenter/snippets/AssetSnippets.java]: Fixed.

File: google-cloud-examples/src/main/java/com/google/cloud/examples/securitycenter/snippets/AssetSnippets.java (hunk 4 above)
Comment (emkornfield) [reply google-cloud-examples/src/main/java/com/google/cloud/examples/securitycenter/snippets/AssetSnippets.java]: Fixed.

File: google-cloud-examples/src/main/java/com/google/cloud/examples/securitycenter/snippets/AssetSnippets.java (hunk 5 above)
Comment (emkornfield) [reply google-cloud-examples/src/main/java/com/google/cloud/examples/securitycenter/snippets/AssetSnippets.java]: I think I got them all, thanks for the thorough review.

File: .kokoro/presubmit/securitycenter-it.cfg (hunk 1 above)
Comment (emkornfield) [reply .kokoro/presubmit/securitycenter-it.cfg]: I was intending this to be for samples, is there a mechanism to do that, or is it just best effort?

**Comment by emkornfield** [comment]
I think I addressed the issues, raised, I'm still unsure of the write approach for automated testing of the snippets (or if it is required)?

**Comment by kolea2** [comment]
Adding @frankyn to answer code example setup questions.

Additionally, @emkornfield - the code format build is failing. Please see https://github.com/googleapis/google-cloud-java/blob/master/CONTRIBUTING.md under 'Coding Style' for how to fix. Thanks!

**Comment by emkornfield** [comment]
@kolea2 thanks, format is fixed.

**Comment by frankyn** [comment]
Hi @emkornfield,

Do you have snippets that you're porting to Java from another language for Cloud Security Command Center that live in Cloud Documentation?

**Comment by emkornfield** [comment]
@frankyn I'm working on redoing the documentation in general.  I've opened https://github.com/googleapis/google-cloud-python/pull/7538 for python which looks slightly different due to API surface area and currently working on Go which I was planning on making these look similar to these.

File: google-cloud-examples/src/main/java/com/google/cloud/examples/securitycenter/snippets/AssetSnippets.java [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+  /**
+   * Lists assets for an organization given meeting <code>filter</code> as of a specific instant in
+   * time.
+   *
+   * @param filter The filter that assets must meet (e.g. {@link #PROJECT_ASSET_FILTERS}). If null,
+   *     all assets in the organization are returned.
+   * @param asOf The instant in time to query for. If null, current time is assumed
+   */
+
+  // [TARGET list_assets]
```
Comment (frankyn) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/securitycenter/snippets/AssetSnippets.java]: I'm not sure if these are needed at all. 
Re: `[TARGET ..] [VARIABLE ..]`

Could you remove them? 

[START ...] Is necessary.

File: google-cloud-examples/src/main/java/com/google/cloud/examples/securitycenter/snippets/AssetSnippets.java [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+  // [VARIABLE "filter"]
+  // [VARIABLE "as_of"]
+  // [START list_assets]
+  public ImmutableList<ListAssetsResult> listAssets(String filter, Instant asOf) {
+    ListAssetsRequest.Builder request =
+        ListAssetsRequest.newBuilder().setParent(organizationName.toString());
```
Comment (frankyn) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/securitycenter/snippets/AssetSnippets.java]: Please add commented variables with a small description of what it's for?
Similar to: https://github.com/googleapis/google-cloud-java/blob/master/google-cloud-examples/src/main/java/com/google/cloud/examples/storage/snippets/StorageSnippets.java#L1459-L1460

File: google-cloud-examples/src/main/java/com/google/cloud/examples/securitycenter/snippets/AssetSnippets.java [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+   * between <code>asOf.minus(timespan)</code> and <code>asOf</code>.
+   *
+   * @param timeSpan The time-range to compare assets over.
+   * @param filter The filter that assets must meet (e.g. {@link #PROJECT_ASSET_FILTERS}). If null,
+   *     all assets in the organization are returned.
+   * @param asOf The instant in time to query for. If null, current time is assumed.
+   */
+  public ImmutableList<ListAssetsResult> listAssetAndStatusChanges(
+      Duration timeSpan, String filter, Instant asOf) {
+    // [START list_asset_changes]
```
Comment (frankyn) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/securitycenter/snippets/AssetSnippets.java]: Please add commented variables with a small description of what it's for?
Similar to: https://github.com/googleapis/google-cloud-java/blob/master/google-cloud-examples/src/main/java/com/google/cloud/examples/storage/snippets/StorageSnippets.java#L1459-L1460

File: google-cloud-examples/src/main/java/com/google/cloud/examples/securitycenter/snippets/AssetSnippets.java (hunk 7 above)
Comment (emkornfield) [reply google-cloud-examples/src/main/java/com/google/cloud/examples/securitycenter/snippets/AssetSnippets.java]: done

File: google-cloud-examples/src/main/java/com/google/cloud/examples/securitycenter/snippets/AssetSnippets.java (hunk 9 above)
Comment (emkornfield) [reply google-cloud-examples/src/main/java/com/google/cloud/examples/securitycenter/snippets/AssetSnippets.java]: I think I documented what you asked for.

File: google-cloud-examples/src/main/java/com/google/cloud/examples/securitycenter/snippets/AssetSnippets.java (hunk 8 above)
Comment (emkornfield) [reply google-cloud-examples/src/main/java/com/google/cloud/examples/securitycenter/snippets/AssetSnippets.java]: I think I documented what you asked for

**Comment by emkornfield** [comment]
@frankyn These aren't staged in documentation yet, since I didn't have a good way to test it (it doesn't appear that I can reference examples until they are merged into master?).  If there is a way could you send it to me internally?

Thanks for the review.

File: .kokoro/presubmit/securitycenter-it.cfg [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+
+# Configure the docker image for kokoro-trampoline.
+env_vars: {
+    key: "TRAMPOLINE_IMAGE"
+    value: "gcr.io/cloud-devrel-kokoro-resources/java8"
+}
+
+env_vars: {
+    key: "INTEGRATION_TEST_ARGS"
+    value: "google-cloud-clients/google-cloud-pubsub"
```
[the remark this answers is not in the document]
Comment (emkornfield) [reply .kokoro/presubmit/securitycenter-it.cfg]: Old, code, I've removed all configuration.

File: google-cloud-examples/src/test/java/com/google/cloud/examples/securitycenter/snippets/ITAssetSnippets.java [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+
+  @AfterClass
+  public static void tearDown() {
+    client.close();
+  }
+
+  private static String getOrganizationId() {
+    return System.getenv("GCLOUD_ORGANIZATION");
+  }
+
```
[the remark this answers is not in the document]
Comment (emkornfield) [reply google-cloud-examples/src/test/java/com/google/cloud/examples/securitycenter/snippets/ITAssetSnippets.java]: Old code?

File: google-cloud-examples/src/test/java/com/google/cloud/examples/securitycenter/snippets/ITAssetSnippets.java [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+  @Test
+  public void testListAssetsWithFilterAndInstance() {
+    assertTrue(3 >= snippets.listAssets(AssetSnippets.PROJECT_ASSET_FILTERS, SOMETHING_INSTANCE).size());
+  }
+
+  @Test
+  public void testChangesReturnsValues() {
+    ImmutableList<ListAssetsResult> result =  snippets.listAssetAndStatusChanges(Duration.ofDays(3), AssetSnippets.PROJECT_ASSET_FILTERS,
```
[the remark this answers is not in the document]
Comment (emkornfield) [reply google-cloud-examples/src/test/java/com/google/cloud/examples/securitycenter/snippets/ITAssetSnippets.java]: Old code?

**Pull request merged**